internal/types/resource/resource.go

```diff
@@ -90,3 +93,17 @@ var Map = map[string]Type{
 	CredentialStore.String():   CredentialStore,
 	CredentialLibrary.String(): CredentialLibrary,
 }
+
+type MinimalInfo struct {
+	PublicId string
+	ScopeId  string
+	UserId   string
+}
+
+type MinimalInfoRepo interface {
+	// Fetches resource IDs for the given scopes. Note that it is assumed that
+	// the set of scopes is either length 1 or is recursive; that is, the reason
+	// we would have multiple scopes is because there is a parent or set of
+	// parents and any children.
+	FetchIdsForScopes(context.Context, []string) (map[string][]MinimalInfo, error)
```

Comment:
Does this need to return a map? It looks like just a `[]MinimalInfo` would be sufficient given how it is used [here](https://github.com/hashicorp/boundary/pull/2042/files#diff-dd4cd0303a44969a76a67e6bcd1024a00b29afcf3bb92ca626b844fac278735eR271-R295)

Reply:
I also think this method could be renamed. It is returning more then just ids, perhaps once a different name for `MinimalInfo` is picked, it would help with this name as well.

Reply:
The map is used in the service handlers to fetch authorized actions based on the resource ID.